### PR TITLE
[macos] simplify "brew_smart_install" helper

### PR DESCRIPTION
**Pull request by ilia-shipitsin**
# Description

simplify "brew_smart_install" helper

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/provision/core/openjdk.sh [hunk 1]
```diff
@@ -29,12 +29,12 @@ installOpenJDK() {
     local JAVA_VERSION=$1
 
     # Get link for Java binaries and Java version
-    assetUrl=$(curl -fsSL "https://api.adoptium.net/v3/assets/latest/${JAVA_VERSION}/hotspot")
+    download_with_retries "https://api.adoptium.net/v3/assets/latest/${JAVA_VERSION}/hotspot" "/tmp" "openjsdk-hotspot.json"
```
Comment (vpolikarpov-akvelon) [inline images/macos/provision/core/openjdk.sh:32]: Did you really mean that name?
```suggestion
    download_with_retries "https://api.adoptium.net/v3/assets/latest/${JAVA_VERSION}/hotspot" "/tmp" "openjdk-hotspot.json"
```

File: images/macos/provision/utils/utils.sh [hunk 2]
```diff
@@ -240,7 +192,18 @@ get_github_package_download_url() {
...
 
-    json=$(curl "${authString[@]}" -fsSL "https://api.github.com/repos/${REPO_ORG}/releases?per_page=${SEARCH_IN_COUNT}")
+    failed=true
+    for i in {1..10}; do
+        curl "${authString[@]}" -fsSL "https://api.github.com/repos/${REPO_ORG}/releases?per_page=${SEARCH_IN_COUNT}" >/tmp/get_github_package_download_url.json && failed=false || sleep 60
+        [ "$failed" = false ] && break
+    done
+
+    if [ "$failed" = true ]; then
+       echo "Failed: brew --cache $dep"
```
Comment (vpolikarpov-akvelon) [inline images/macos/provision/utils/utils.sh:202]: Seems like you forgot to update error message text here

Comment (ilia-shipitsin) [reply images/macos/provision/utils/utils.sh:202]: thanks, fixed

File: images/macos/provision/core/openjdk.sh [hunk 3]
```diff
@@ -29,12 +29,12 @@ installOpenJDK() {
     local JAVA_VERSION=$1
 
     # Get link for Java binaries and Java version
-    assetUrl=$(curl -fsSL "https://api.adoptium.net/v3/assets/latest/${JAVA_VERSION}/hotspot")
+    download_with_retries "https://api.adoptium.net/v3/assets/latest/${JAVA_VERSION}/hotspot" "/tmp" "openjdk-hotspot.json"
 
     if [[ $arch == "arm64" ]]; then
-        asset=$(echo ${assetUrl} | jq -r '.[] | select(.binary.os=="mac" and .binary.image_type=="jdk" and .binary.architecture=="aarch64")')
+        asset=$(jq -r '.[] | select(.binary.os=="mac" and .binary.image_type=="jdk" and .binary.architecture=="aarch64")' /tmp/openjsdk-hotspot.json)
```
Comment (vpolikarpov-akvelon) [inline images/macos/provision/core/openjdk.sh:35]:
```suggestion
        asset=$(jq -r '.[] | select(.binary.os=="mac" and .binary.image_type=="jdk" and .binary.architecture=="aarch64")' /tmp/openjdk-hotspot.json)
```

File: images/macos/provision/core/openjdk.sh [hunk 4]
```diff
@@ -29,12 +29,12 @@ installOpenJDK() {
...
     else
-        asset=$(echo ${assetUrl} | jq -r '.[] | select(.binary.os=="mac" and .binary.image_type=="jdk" and .binary.architecture=="x64")')
+        asset=$(jq -r '.[] | select(.binary.os=="mac" and .binary.image_type=="jdk" and .binary.architecture=="x64")' /tmp/openjsdk-hotspot.json)
```
Comment (vpolikarpov-akvelon) [inline images/macos/provision/core/openjdk.sh:37]:
```suggestion
        asset=$(jq -r '.[] | select(.binary.os=="mac" and .binary.image_type=="jdk" and .binary.architecture=="x64")' /tmp/openjdk-hotspot.json)
```

**Review by vpolikarpov-akvelon — CHANGES_REQUESTED** [review]

**Review by vpolikarpov-akvelon — APPROVED** [review]

**Review by mikhailkoliada — APPROVED** [review]

**Pull request merged**